org.adempiere.base/src/org/compiere/model/MLookup.java

```diff
@@ -995,6 +996,148 @@ private synchronized static CCache<String, ValueNamePair> getDirectValueNamePair
 		return vnpCache;
 	}
 	
+
+
+	/**
+	 * Get Lookup
+	 * @param tableID
+	 * @param windowNo
+	 * @param tabNo
+	 * @return null if tableID &lt;= 0 or the table doesn't have any key column, else {@link MLookup}
+	 */
+	public static MLookup getRecordsLookup(int tableID, int windowNo, int tabNo) {
+		return getRecordsLookup(tableID, false, windowNo, tabNo);
+	}
+	
+	/**
+	 * Get Lookup
+	 * @param tableID
+	 * @param windowNo
+	 * @param tabNo
+	 * @param keyColumn
+	 * @return null if tableID &lt;= 0 or the table doesn't have any key column, else {@link MLookup}
+	 */
+	public static MLookup getRecordsLookup(int tableID, int windowNo, int tabNo, String keyColumn) {
+		return getRecordsLookup(tableID, false, windowNo, tabNo, keyColumn);
+	}
+
+	/**
+	 * Get Lookup
+	 * @param tableID
+	 * @param tableBased - if the lookup checks for the UUID Key Column based on the table (true), or on the columnName (false)
+	 * @param windowNo
+	 * @param tabNo
+	 * @return null if tableID <= 0 or the table doesn't have any key column, else {@link MLookup}
+	 */
+	public static MLookup getRecordsLookup(int tableID, boolean tableBased, int windowNo, int tabNo) {
+		return getRecordsLookup(tableID, tableBased, windowNo, tabNo, null);
+	}
+	
+	/**
+	 * Get Lookup
+	 * @param tableID
+	 * @param tableBased - if the lookup checks for the UUID Key Column based on the table (true), or on the columnName (false)

```

Comment:
This fail to mention this is ignore if keyColumn argument is not null.
Also, it got nothing to do with whether the method is "Table Based", from what I've goes through below, it is more like whether it is using a table's integer id key column (if available) or UUID key column. In other word, both scenario is "table based".